docs/can-guides/commitment/recipes/credit-card-advanced/9-disable-payments.js

```diff
@@ -1,187 +1,266 @@
 
-// HELPER FUNCTIONS
-function validateCard(card) {
-	if (!card) {
-		return "There is no card"
-	}
-	if (card.length !== 16) {
-		return "There should be 16 characters in a card";
-	}
-}
+		get card() {
+			return Kefir.combine([this.cardNumber, this.expiry, this.cvc],
+				(cardNumber, expiry, cvc) => {
+					return {cardNumber , expiry , cvc};
+			});
+		},
+		
+		// STREAM< Promise<Number> | undefined >
+		get paymentPromises(){
+			return Kefir.combine([this.payClicked], [this.card], (payClicked, card) => {
+				if (payClicked) {
+					console.log("Asking for token with", card);
+					return new Promise((resolve) => {
+						setTimeout(() => {
+							resolve(1000);
+						}, 2000);
+					})
+				}
+			});
+		},
 
-function validateExpiry(expiry) {
-	if (!expiry) {
-		return "There is no expiry. Format  MM-YY";
-	}
-	if (expiry.length !== 2 || expiry[0].length !== 2 || expiry[1].length !== 2) {
-		return "Expirty must be formatted like MM-YY";
-	}
-}
+		// STREAM< STREAM<STATUS> >
+		// This is a stream of streams of status objects.
+		get paymentStatusStream () {
+			return this.paymentPromises.map((promise) => {
+				if (promise) {
+					// STREAM<STATUS>
+					return Kefir.concat([
+						Kefir.constant({
+							status: "pending"
+						}),
+						Kefir.fromPromise(promise).map((value) => {
+							return {
+								status: "resolved",
+								value: value
+							};
+						})
+					]);
+				} else {
+					// STREAM
+					return Kefir.constant({
+						status: "waiting"
+					});
+				}
+			});
+		},
 
-function validateCVC(cvc) {
-	if (!cvc) {
-		return "There is no CVC code";
-	}
-	if (cvc.length !== 3) {
-		return "The CVC must be at least 3 numbers";
-	}
-	if (isNaN(parseInt(cvc))) {
-		return "The CVC must be numbers";
-	}
-}
+		// STREAM<STATUS> //{status: "waiting"} | {status: "resolved"}
+		get paymentStatus(){
+			return this.paymentStatusStream.flatMap().toProperty();
+		},
+
+		get disablePaymentButton() {
+			return Kefir.combine([this.isCardInvalid, this.paymentStatus],
+				(isCardInvalid, paymentStatus) => {
+					return (isCardInvalid === true) || !paymentStatus || paymentStatus.status === "pending";
+				}).toProperty(() => {
+					return true;
+				});
+		},
+
+		pay(event) {
+			event.preventDefault();
+			this.payClicked.emitter.value(true)
+		},
 
-function showOnlyWhenBlurredOnce(errorStream, blurredStream) {
-	const errorEvent = errorStream.map((error) => {
-		if (!error) {
-			return {
-				type: "valid"
+		// HELPER FUNCTIONS
+		validateCard(card) {
+			if (!card) {
+				return "There is no card"
 			}
-		} else {
-			return {
-				type: "invalid",
-				message: error
+			if (card.length !== 16) {
+				return "There should be 16 characters in a card";
 			}
-		}
-	});
+		},
+		validateExpiry(expiry) {
+			if (!expiry) {
+				return "There is no expiry. Format  MM-YY";
+			}
+			if (expiry.length !== 2 || expiry[0].length !== 2 || expiry[1].length !== 2) {
+				return "Expirty must be formatted like MM-YY";
```

Comment:
Expirty -> Expiry (here and in other places)

Reply:
Fixed